src/teams.rs

```diff
@@ -4,84 +4,232 @@ use std::collections::BTreeMap;
+    // If this is intended, document why.
+    name: String,
+    ping: String,
+    members: Vec<String>,
+}
+
+impl Team {
+    pub fn member_logins(&self) -> impl Iterator<Item = &str> {
+        self.members.iter().map(|s| s.as_str())
+    }
 }
 
 #[derive(Debug, Eq, Hash, Ord, PartialEq, PartialOrd)]
 pub struct TeamLabel(pub String);
 
-type TeamsFromFile = BTreeMap<String, TeamFromFile>;
-pub type Teams = BTreeMap<TeamLabel, Team>;
+impl<'de> de::Deserialize<'de> for TeamLabel {
+    fn deserialize<D: de::Deserializer<'de>>(de: D) -> Result<Self, D::Error> {
+        String::deserialize(de).map(TeamLabel)
+    }
 
-#[derive(Debug, Deserialize)]
-struct TeamFromFile {
-  name: String,
-  ping: String,
-  members: Vec<String>,
+    fn deserialize_in_place<D: de::Deserializer<'de>>(de: D, place: &mut Self)
+        -> Result<(), D::Error>
+    {
+        String::deserialize_in_place(de, &mut place.0)
+    }
+}
+
+//==============================================================================
+// Implementation details
+//==============================================================================
+
+/// Read the validated `rfcbot.toml` configuration file.
+fn read_rfcbot_cfg_validated() -> RfcbotConfig {
+    let cfg = read_rfcbot_cfg();
+
+    cfg.teams.values().for_each(|team|
+        team.validate()
+            .expect("unable to verify team member from database.
+if you're running this for tests, make sure you've pulled github users from prod")
+    );
+
+    cfg
+}
+
+/// Read the unprocessed `rfcbot.toml` configuration file.
+fn read_rfcbot_cfg() -> RfcbotConfig {
+    read_rfcbot_cfg_from(
+        include_str!(concat!(env!("CARGO_MANIFEST_DIR"), "/rfcbot.toml")))
+}
+
+fn read_rfcbot_cfg_from(input: &str) -> RfcbotConfig {
+    toml::from_str(input).expect("couldn't parse rfcbot.toml!")
 }
 
-impl TeamFromFile {
-  pub fn validate(self) -> DashResult<Team> {
-    use domain::schema::githubuser::dsl::*;
-    let conn = &*(DB_POOL.get()?);
-
-    // bail if they don't exist, but we don't want to actually keep the id in ram
-    for member_login in &self.members {
-      match githubuser
-        .filter(login.eq(member_login))
-        .first::<GitHubUser>(conn) {
-            Ok(_) => (),
-            Err(why) => {
-                error!("unable to find {} in database: {:?}", member_login, why);
-                return Err(why.into());
+impl Team {
+    fn validate(&self) -> DashResult<()> {
+        use domain::schema::githubuser::dsl::*;
+        let conn = &*(DB_POOL.get()?);
+
+        // bail if they don't exist, but we don't want to actually keep the id in ram
+        for member_login in self.member_logins() {
+            match githubuser
+                    .filter(login.eq(member_login))
+                    .first::<GitHubUser>(conn)
+            {
+                Ok(_) => (),
+                Err(why) => {
+                    error!("unable to find {} in database: {:?}", member_login, why);
```

Comment:
This makes me think switching to `failure` would be nice, would just be a single line.

Reply:
Yeah I think we need to do something about these at some point; but I'd like to follow up with cleanup at a later point since I'm not very familiar with `failure` right now ;)